// docpad-plugin-coffeelint/out/coffeelint.plugin.js
// Generated by CoffeeScript 1.6.3
var __hasProp = {}.hasOwnProperty,
  __extends = function(child, parent) { for (var key in parent) { if (__hasProp.call(parent, key)) child[key] = parent[key]; } function ctor() { this.constructor = child; } ctor.prototype = parent.prototype; child.prototype = new ctor(); child.__super__ = parent.prototype; return child; };

module.exports = function(BasePlugin) {
  var CoffeeLintPlugin, coffeelint, colors, _ref;
  coffeelint = require('coffeelint');
  colors = require('colors');
  return CoffeeLintPlugin = (function(_super) {
    __extends(CoffeeLintPlugin, _super);

    function CoffeeLintPlugin() {
      _ref = CoffeeLintPlugin.__super__.constructor.apply(this, arguments);
      return _ref;
    }

    CoffeeLintPlugin.prototype.name = 'coffeelint';

    CoffeeLintPlugin.prototype.config = {
      ignorePaths: [],
      ignoreFiles: [],
      lintOptions: {}
    };

    CoffeeLintPlugin.prototype.renderBefore = function(_arg) {
      var collection, config, ignoredPaths, maxErrors,
        _this = this;
      collection = _arg.collection;
      if (docpad.getEnvironment() === 'development') {
        config = this.config;
        ignoredPaths = [];
        if (config.lintOptions.maxerr) {
          maxErrors = config.lintOptions.maxerr;
        } else {
          maxErrors = 50;
        }
        config.ignorePaths.map(function(path, i) {
          path = path.toString();
          if (path.charAt(0) === '/') {
            path.slice(1);
          }
          if (path.charAt(path.length - 1) !== '/') {
            path = path + '/';
          }
          return ignoredPaths.push(path);
        });
        return collection.each(function(item) {
          var err, file, fileName, message, path, ref, _i, _j, _k, _len, _len1, _len2, _ref1, _ref2;
          file = item.attributes;
          if (file.extension === 'coffee') {
            for (_i = 0, _len = ignoredPaths.length; _i < _len; _i++) {
              path = ignoredPaths[_i];
              if (file.relativePath.indexOf(path) === 0) {
                return;
              }
            }
            _ref1 = config.ignoreFiles;
            for (_j = 0, _len1 = _ref1.length; _j < _len1; _j++) {
              fileName = _ref1[_j];
              if (file.relativePath === fileName) {
                return;
              }
            }
            if (coffeelint.lint(file.body, config.options).length === 0) {

            } else {
              console.log('CoffeeLint - '.white + file.relativePath.red);
              coffeelint.errors = coffeelint.lint(file.body, config.options);
              _ref2 = coffeelint.errors;
              for (_k = 0, _len2 = _ref2.length; _k < _len2; _k++) {
                err = _ref2[_k];
                ref = 'line ' + err.lineNumber;
                message = err.message;
                console.log(ref.blue + ' - '.white + message);
              }
              return console.log('\n');
            }
          }
        });
      }
    };

    return CoffeeLintPlugin;

  })(BasePlugin);
};
